backend/src/main/java/pl/edu/agh/backend/auth/AuthenticationController.java

```diff
@@ -0,0 +1,37 @@
+package pl.edu.agh.backend.auth;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import pl.edu.agh.backend.auth.models.AuthenticationRequest;
+import pl.edu.agh.backend.auth.models.AuthenticationResponse;
+import pl.edu.agh.backend.auth.models.RegisterRequest;
+
+@RestController
+@RequestMapping("/api/v1/auth")
+@RequiredArgsConstructor
+public class AuthenticationController {
+
+    private final AuthenticationService authenticationService;
+
+    @PostMapping("/register")
+    public ResponseEntity<AuthenticationResponse> register(
+            @RequestBody RegisterRequest request
+    ) {
+        return authenticationService.register(request)
+                .map(ResponseEntity::ok)
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_ACCEPTABLE));
+    }
+
+    @PostMapping("/authenticate")
+    public ResponseEntity<AuthenticationResponse> authenticate(
```

Comment:
to samo, register i login są bardziej intuicyjne

Reply:
sign-up, sign-in?